taskcollector/TaskCollector.cpp

```diff
@@ -1,81 +1,78 @@
 #include "FreeImage.h"
+#include <algorithm>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <list>
+#include <memory>
 #include <queue>
 #include <iostream>
 
-
-std::queue<FIBITMAP *> alphaChunks;	
+using bitmap_t = std::shared_ptr<FIBITMAP>;
 
 /** Generic image loader
-	@param lpszPathName Pointer to the full file name
-	@param flag Optional load flag constant
-	@return Returns the loaded dib if successful, returns NULL otherwise
+@param lpszPathName Pointer to the full file name
+@param flag Optional load flag constant
+@return Returns the loaded dib if successful, returns NULL otherwise
 */
-FIBITMAP* GenericLoader(const char* lpszPathName, int flag) {
+bitmap_t GenericLoader(const char* lpszPathName, int flag) {

```

Comment:
I would keep information that it is still a pointer. How about naming it `bitmap_ptr`?